Python/HAT/hypergraph.py

```
@@ -0,0 +1,575 @@
+            warnings.warn('The provided and detected `uniform` are not in agreement!')
+
+        if self._order != order and order is not None:
+            warnings.warn('The provided and detected `order` are not in agreement!')
+
+    def _detect_directed(self, directed):
+        if self._edges is not None and 'Head' in self._edges.columns and 'Tail' in self._edges.columns:
+            self._directed = True
+
+        # The adjacency tensor is directed
+        elif self._adjacency_tensor is not None:
+            self._directed = True
```

Ruff (0.8.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)